### ISSUE TITLE
PHP 8.1 does not appear to work

### ISSUE THREAD

**Issue by jSML4ThWwBID69YC** [unbit/uwsgi#2394]
I was testing PHP 8.1 with uWSGi and it does not appear to work. 

The plugin builds with the following error. 
```
 # ./uwsgi --build-plugin "plugins/php php"
*** uWSGI building and linking plugin from plugins/php ***
[cc -pthread] php_plugin.so
plugins/php/php_plugin.c:1049:30: warning: incompatible pointer types assigning to 'zend_string *' (aka 'struct _zend_string *') from 'char [1025]' [-Wincompatible-pointer-types]
        file_handle.filename = real_filename;
                             ^ ~~~~~~~~~~~~~
1 warning generated.
build time: 0 seconds
*** php plugin built and available in php_plugin.so ***
```

I tested the PHP installation from the command line and that works. Running it through the uWSGI plugin returns a 500 error code. Even simple scripts like hello world, or phpinfo fail. 

The php error log shows this. 

```
[12-Jan-2022 18:10:26 UTC] PHP Fatal error:  Uncaught ValueError: Path cannot be empty in [no active file]:0
Stack trace:
#0 {main}
  thrown in [no active file] on line 0
````

OS: FreeBSD
uWSGI: v2.0.20
PHP: 8.1.1 from the FreeBSD ports tree. 

Any ideas on how to fix this?

**Comment by dvzrv** [comment]
@xrmx Could you have a look at this? It is currently also broken for Arch Linux it seems (e.g. https://bugs.archlinux.org/task/73470 and https://bugs.archlinux.org/task/73452#comment205218).

**Comment by gwstorm** [comment]
I have a real-world example of this. I use uwsgi for Nextcloud and Nextcloud is working, however, it's unreachable and the fatal error is a truncated 'loud/index.php' instead of 'nextcloud/index.php'

Full error -

Jan 23 21:24:53 server uwsgi[81310]: [pid: 81310|app: -1|req: -1/1] 10.0.0.10 () {56 vars in 1015 bytes} [Sun Jan 23 21:24:53 2022] GET / => generated 0 bytes in 1 msecs (HTTP/2.0 500) 2 headers in 103 bytes (0 switches on core 0)
Jan 23 21:24:53 server uwsgi[81310]: PHP Fatal error: Failed opening required 'loud/index.php' (include_path='.:') in Unknown on line 0
Jan 23 21:24:53 server uwsgi[81310]: PHP Warning: Unknown: Failed to open stream: No such file or directory in Unknown on line 0

**Comment by yu17** [comment]
I have encountered the exact same issue as @gwstorm did.

I have both nextcloud and cacti setup with uwsgi, and ran into the following issue.
Nextcloud:
```
uwsgi[1558]: PHP Fatal error:  Failed opening required 'loud/index.php' (include_path='.:') in Unknown on line 0
```
Cacti:
```
uwsgi[1893821]: PHP Fatal error:  Failed opening required '/index.php' (include_path='.:') in Unknown on line 0
```

The correct path in these cases should have been `/usr/share/webapps/nextcloud/index.php` and `/usr/share/webapps/cacti/index.php` respectively. Apparently, in both cases, exactly 24 characters were truncated from the path. A pretty funny bug.

**Comment by jSML4ThWwBID69YC** [comment]
The patch works for me. Tested with PHP 8.1 and WordPress 5.9.

Thank you @xrmx

**Comment by xrmx** [comment]
Should be fixed in 2.0.21.
